app-collab-note/src/utils/DataApiUtils/bridge.js

```diff
@@ -1,150 +1,145 @@
-      throw new Error(msg);
-    }
+  query.token = authToken_;
 
-    if(responseType === 'blob') {
-      const blob = await response.blob();
-      return blob;
+  let queryString = '';
+  if (query) {
+    const s = [];
+    for (const k in query) {
+      if (!Object.prototype.hasOwnProperty.call(query, k)) continue;
+      s.push(`${encodeURIComponent(k)}=${encodeURIComponent(query[k])}`);
     }
-
-    const json = await response.json();
-    return json;
+    queryString = s.join('&');
+    if (queryString) queryString = `?${queryString}`;
   }
 
-  async updateNoteContent(note) {
-    const { id, body } = note;
-    if (!id) throw new Error('Cannot update a note without id');
-    return this.clipperApiExec('PUT', `notes/${id}`, {}, { body });
+  const response = await fetch(`${baseUrl}/${path}${queryString}`, fetchOptions);
+  if (!response.ok) {
+    const msg = await response.text();
+    throw new Error(msg);
   }
 
-  async getNote(id, fields = []) {
-    if (!id) throw new Error('Cannot get a note without id');
-    return this.clipperApiExec('GET', `notes/${id}`, { fields: fields.join(',') });
+  if(responseType === 'blob') {
+    const blob = await response.blob();
+    return blob;
   }
 
-  async getNoteResouceListWithBlob(noteId) {
-    if (!noteId) throw new Error('Cannot get resource list without note id');
-    const resp = await this.clipperApiExec('GET', `notes/${noteId}/resources`, {});
+  const json = await response.json();
+  return json;
+}
 
-    const resourceList = resp.items || [];
-    for(const i in resourceList) {
-      try {
-        const blob = await this.getResouceFile(resourceList[i].id);
-        resourceList[i].dataURI = await this.getDataURI(blob);
-      }
-      catch(error) {
-        console.warn(`Error while fetching resource file: ${error.message}`);
-      }
-    }
-    
-    return resourceList;
-  }
+async function updateNoteContent(note) {
+  const { id, body } = note;
+  if (!id) throw new Error('Cannot update a note without id');
+  return clipperApiExec('PUT', `notes/${id}`, {}, { body });
+}
 
-  async getResouceFile(resourceId) {
-    if (!resourceId) throw new Error('Cannot get resource file without resource id');
-    return this.clipperApiExec('GET', `/resources/${resourceId}/file`, {}, null, 'blob');
-  }
+async function getNote(id, fields = []) {
+  if (!id) throw new Error('Cannot get a note without id');
+  return clipperApiExec('GET', `notes/${id}`, { fields: fields.join(',') });
+}
 
-  getDataURI(blob) {
-    return new Promise((resolve, reject) => {
-      const reader = new FileReader();
-      reader.onload = (e) => resolve(reader.result);
-      reader.onerror = reject;
-      reader.readAsDataURL(blob);
-    });
+async function getNoteResouceList(noteId) {
+  if (!noteId) throw new Error('Cannot get resource list without note id');
+  const resp = await clipperApiExec('GET', `notes/${noteId}/resources`, {});
+
+  const resourceList = resp.items || [];
```

Comment:
Ideally you should handle pagination here (or better in `clipperApiExec`). However it's not very likely to find a note with > 100 resources so I suppose this can be done later.